Shop.DAL/Interfaces/IGameRepository.cs

```diff
@@ -0,0 +1,16 @@
+using Shop.DAL.Common.RequestParams;
+using Shop.DAL.Models;
+
+namespace Shop.DAL.Interfaces;
+
+public interface IGameRepository
+{
+    Task<IEnumerable<Game>> GetGamesAsync(GameParameters parameters,
+        CancellationToken cancellationToken);
+    Task<Game> GetGameByIdAsync(Guid id, CancellationToken cancellationToken);
+    Task<IEnumerable<Game>> GetGamesByCategoryAsync(Category category, GameParameters parameters,
+        CancellationToken cancellationToken);
```

Comment:
create a basic repository and make a unique GetBySmth type method using a predicate